frontend/src/pages/RoomPage.tsx

```diff
@@ -257,27 +300,110 @@ export const RoomPage = () => {
+              textTransform: "none",
+            }}
+            icon={<Wysiwyg />}
+            iconPosition="start"
+          />
+          <Tab
+            label="Submission"
+            value="submission"
+            sx={{
+              fontWeight: "bold",
+              fontSize: "1.2rem",
+              textTransform: "none",
+            }}
+            icon={<DriveFolderUpload />}
+            iconPosition="start"
+          />
+        </TabList>
+        <TabPanel sx={{ p: 0 }} value="description">
+          <Stack
+            direction="row"
+            spacing={2}
+            sx={{ width: "100%", flex: 1, minHeight: 0, p: 3 }}
+          >
+            <Stack spacing={2} sx={{ minWidth: "40%", maxWidth: "40%" }}>
+              <Box sx={{ flex: 1, minHeight: 0 }}>
+                <Question />
+              </Box>
+              <ChatProvider roomId={roomId || ""}>
+                <Box sx={{ height: "40%" }}>
+                  <Chat />
+                </Box>
+              </ChatProvider>
+            </Stack>
+            <Box sx={{ flex: 1, minWidth: 0 }}>
+              <EditorProvider roomId={roomId || ""}>
+                <Editor language={"javascript"} />
+              </EditorProvider>
             </Box>
-          </ChatProvider>
-        </Stack>
-        <Box sx={{ height: "100%", flex: 1, minWidth: 0 }}>
-          <EditorProvider roomId={roomId || ""}>
-            <Editor language={"javascript"} />
-          </EditorProvider>
-        </Box>
-      </Stack>
+          </Stack>
+          <Stack
+            direction="row"
+            justifyContent="flex-end"
+            sx={{ py: 2, px: 3 }}
+          >
+            <StyledButton
+              label={"Submit Code"}
+              sx={{ "&:hover": { boxShadow: "1" } }}
+            />
+          </Stack>
+        </TabPanel>
+        <TabPanel sx={{ p: 0 }} value="submission">
+          <Stack
+            direction="row"
+            spacing={2}
+            sx={{ width: "100%", flex: 1, minHeight: 0, p: 3 }}
+          >
+            <TableContainer component={Paper}>
+              <Table sx={{ minWidth: "100%" }}>
+                <TableHead sx={{ bgcolor: "primary.500" }}>
+                  <TableRow>
+                    {tableHeaders.map((tableHeader) => (
+                      <TableCell
+                        key={tableHeader}
+                        align="center"
+                        sx={{
+                          fontWeight: "bold",
+                          color: "white",
+                        }}
+                      >
+                        {tableHeader}
+                      </TableCell>
+                    ))}
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  <TableRow>
+                    {tableCells.map((tableCell) => (
+                      <TableCell key={tableCell} align="center">
+                        {tableCell}
```

Comment:
Should we have a separate TableCell for Pass, Fail, Compile Error? So that we can style the table cell values of the judgement

Reply:
We can use an enum and style the table cells based on the value. We might want to consider one more value for TLE.